### PR TITLE
connectors-ci: bust cache on java connector build

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Relates to https://github.com/airbytehq/airbyte/issues/28299

We observe that some java connectors are occasionally built with stale code.

## How
* Bust the cache on connector container build by introducing a cache buster env var.
* Use `with_new_file` to copy the connector tar archive to the container

### CODE REVIEW

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py
Comment (evantahler) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:787]: non-blocking question - if every java build makes a new cache 'namespace' with this random UUID... do we ever clean up this chache?  It seems like it could get very large

**Review by evantahler — APPROVED** [review]

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py
Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:787]: My understanding is that once the env var is set all the layers afterward are busted. I believe that once busted the layers are not cached anymore and the garbage collection logic is on the build kit side.

Comment (alafanechere) [reply tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py:787]: And, even if the layers are not garbage collected, this change won't introduce uuid namespaced cache as layers are not namespaced and are reused / shared across images, so we're not introducing an ever growing cache.

**Review by bnchrch — APPROVED** [review]
Ill approve now, but lets not merge this unless we know Daggers fix does not work

**Pull request closed without merging**
